convert2rhel/actions/__init__.py

```diff
@@ -206,25 +211,94 @@ def run(self):
 
         self._has_run = True
 
-    def set_result(self, status=_NO_USER_VALUE, error_id=_NO_USER_VALUE, message=_NO_USER_VALUE):
+    @property
+    def result(self):
+        return self._result
+
+    @result.setter
+    def result(self, action_message):
+        """
+        Make sure the correct policy for result messages is used.
+        """
+        if not isinstance(action_message, ActionResult):
+            raise Exception()
+
+        self._result = action_message
+
+    def set_result(self, level, id=None, message=None):
+        """
+        Helper method that sets the resulting values for level, id and message.
+
+        :param id: The id to identify the result.
+        :type id: str
+        :param level: The status_code of the result .
+        :type: level: str
+        :param message: The message to be set.
+        :type message: str | None
+        """
+        if level not in ("ERROR", "OVERRIDABLE", "SKIP", "SUCCESS"):
+            raise KeyError("The level of result must be FAILURE, OVERRIDABLE, SKIP, or SUCCESS.")
+
+        result = ActionResult(level, id, message)
+        self.result = result
+
+    def add_message(self, level, id=None, message=None):
         """
-        Helper method that sets the resulting values for status, error_id and message.
+        Helper method that adds the resulting values for level, id and message of a warning or info log message.
 
-        :param status: The status to be set.
-        :type: status: str
-        :param error_id: The error_id to identify the error.
-        :type error_id: str
+        :param id: The id to identify the message.
+        :type id: str
+        :param level: The level to be set.
+        :type: level: str
         :param message: The message to be set.
         :type message: str | None
         """
-        if status != _NO_USER_VALUE:
-            self.status = STATUS_CODE[status]
+        msg = ActionMessage(level, id, message)
+        self.messages.append(msg)
 
-        if error_id != _NO_USER_VALUE:
-            self.error_id = error_id
 
-        if message != _NO_USER_VALUE:
-            self.message = message
+class ActionMessageBase:
+    def __init__(self, level="SUCCESS", id=None, message=""):
+        self.id = id
+        self.level = STATUS_CODE[level]
+        self.message = message
+
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "level": self.level,
+            "message": self.message,
+        }
+
+
+class ActionMessage(ActionMessageBase):
+    def __init__(self, level, id=None, message=""):
+        if not (id and level and message):
+            raise InvalidMessage("Messages require id, level and message fields")
```

Comment:
Since all three are required at this point, I believe you could remove the assigment for the param. 

```suggestion
    def __init__(self, level, id, message):
        if not (id and level and message):
            raise InvalidMessage("Messages require id, level and message fields")
```